api: add v1 API protobuf definition and generated codes.

yifan-gu opened the pull request:
/cc @jonboulle @eyakubovich @philips

yifan-gu commented:
Major changes from the original proposal is:
- Renaming `GetPodStatus` to `InspectPod`
- Remove `PodStatus` type, merge its fields into `Pod` type.

yifan-gu commented:
Related to #1193 #1208

Also cc our kubelet friends :) @yujuhong @dchen1107 @vishh

vishh commented on api/v1/api.proto:
```diff
@@ -0,0 +1,314 @@
+// PodState defines the possible states of the pod.
+enum PodState {
+        POD_STATE_UNDEFINED       = 0;
+        POD_STATE_EMBRYO          = 1;
```
Can you document the semantics of each of these states? I assume the kubelet will have to directly deal with these states.

vishh commented on api/v1/api.proto:
```diff
@@ -0,0 +1,314 @@
+        // Type of the event.
+        EventType type = 1;
+
+        // ID of the subject that causes the event, it can be the pod id or the image id.
```
What is the plan to communicate individual container lifecycle events? This will be required for updating [ContainerStatus](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L1080) in k8s API.

xiang90 commented on api/v1/api.proto:
```diff
@@ -0,0 +1,314 @@
+// Copyright 2014 The rkt Authors
```
2015?

yifan-gu commented:
@vishh Added app state events and documented the pod states.

yifan-gu commented:
Added one more rpc, `AddEvent` which will be invoked by rkt pod to add events into the event list.

yifan-gu commented:
Updated the `Pod` type: removed `name`, added `manifest`.
Also added `annotation_requirements` in `PodFilter` and `ImageFilter` so that user can use this as label selectors like in k8s.

yifan-gu commented:
@jonboulle I implemented all the intefaces in the my [api_service branch](https://github.com/coreos/rkt/pull/1411/commits). Seems it's time to get those merged one by one now.

yifan-gu commented:
Anyone want to give a lgtm? @jonboulle @iaguis @alban

jonboulle commented:
@yifan-gu could you send an email to rkt-dev asking for feedback?

alban commented on api/v1/api.pb.go:
```diff
@@ -0,0 +1,988 @@
+// Code generated by protoc-gen-go.
+// source: api.proto
+// DO NOT EDIT!
```
Since this is a generated file, shouldn't it be removed from the git repository and add calls to `protoc-gen-go` in some makefiles?

iaguis replied:
It's probably needed by apps that want to build against it.

yifan-gu replied:
yes

alban commented on api/v1/api.proto:
```diff
@@ -0,0 +1,395 @@
+        string rkt_version = 1;
+
+        // Version of appc, required, in the form of Semantic Versioning (http://semver.org/).
+        string appc_version = 2;
```
Should the version of this event stream protocol be added there? `v1` at the moment...

yifan-gu replied:
The user should be aware of the version when using the grpc client interface. but it doesn't cost anything to add it here :)

yifan-gu commented:
Addressed @alban 's comments. Wonder if @thockin is interested in this and will be glad to hear some feedback :)

yifan-gu commented:
Will merge after CI green if no more objections

jonboulle commented:
As discussed oob please mark this clearly as alpha and document that it is
experimental and should not be used
On Sep 28, 2015 11:40 AM, "Yifan Gu" <notifications@github.com> wrote:

> Will merge after CI green if no more objections
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/coreos/rkt/pull/1359#issuecomment-143838108>.
>

yifan-gu commented:
> As discussed oob please mark this clearly as alpha and document that it is
experimental and should not be used

Marked in the .proto file here:
https://github.com/yifan-gu/rocket/commit/28968f58522cb15a0d671d411562b264c85382b4#diff-1f6fbc35b4762277484db51a76548ad5R16
